diff --git a/cmake/cmake_extension.py b/cmake/cmake_extension.py
index ae27abf..b812994 100644
--- a/cmake/cmake_extension.py
+++ b/cmake/cmake_extension.py
@@ -53,6 +53,7 @@ class BuildExtension(build_ext):
 
         # build/lib.linux-x86_64-3.8
         os.makedirs(self.build_lib, exist_ok=True)
+        self.build_temp = "build_release"
 
         kaldifeat_dir = Path(__file__).parent.parent.resolve()
 
@@ -71,17 +72,17 @@ class BuildExtension(build_ext):
             print(f"Setting PYTHON_EXECUTABLE to {sys.executable}")
             cmake_args += f" -DPYTHON_EXECUTABLE={sys.executable}"
 
-        build_cmd = f"""
-            cd {self.build_temp}
+        #  build_cmd = f"""
+            #  cd {self.build_temp}
 
-            cmake {cmake_args} {kaldifeat_dir}
+            #  cmake {cmake_args} {kaldifeat_dir}
 
+            #  make {make_args} _kaldifeat
+        #  """
+        #  print(f"build command is:\n{build_cmd}")
 
-            make {make_args} _kaldifeat
-        """
-        print(f"build command is:\n{build_cmd}")
-
-        ret = os.system(build_cmd)
+        #  ret = os.system(build_cmd)
+        ret = 0
         if ret != 0:
             raise Exception(
                 "\nBuild kaldifeat failed. Please check the error message.\n"
